cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -670,6 +729,17 @@ def _create_backing(self, volume, host=None, create_params=None):
                                                 extra_config=extra_config)
 
         self.volumeops.update_backing_disk_uuid(backing, volume['id'])
+        if (self.configuration.vmware_datastores_as_pools and
+                self._is_pool_outdated_for_volume(volume)):
+            # TODO(walt) - this writes the volume update to the db. :(
+            # This sucks, but don't have any other way
```

Comment:
Why don't we trigger here a migration as in the other cases when we change the host?
I know, it will run on the same agent, but shouldn't that avoid writing directly in the db in the driver?

Reply:
There is no need to 'migrate'.   The volume didn't have any backing previously and now does, because of the lazy create feature.   All this does is update the host entry, because it now has a pool/datastore assigned to it where it didn't previously.

Reply:
That we do not have a volume in the backend is irrelevant to the higher layers. That is an internal property of the driver, and will make the migration all the more easy.

The method to move a volume from one backend to another is simply called migrate, and the way I see it, if we use those abstractions, there is no need to modify the model from within the driver.

Reply:
It's way too late at this point.  The user didn't want a migrate, they wanted an attach or a clone or a backup.  There was no backing, and now there is and the host has to be set correctly when pools are enabled.  this is a completely different path than what you are suggesting.   Migrate does not apply.